pkg/generate/config/resource.go

```diff
@@ -349,3 +353,19 @@ func (c *Config) ResourceShortNames(resourceName string) []string {
 	}
 	return rConfig.ShortNames
 }
+
+// ResourceIsAdoptable returns whether the given CRD is adoptable
+func (c *Config) ResourceIsAdoptable(resourceName string) bool {
+	if c == nil {
+		return true
+	}
+	rConfig, ok := c.Resources[resourceName]
+	if !ok {
+		return true
+	}
+	// Default to True
```

Comment:
I'm wondering actually if we should default to `false` instead. My reasoning is two-fold. First, the zero value of a `bool` is false, so an "empty" `is_adoptable` field in the generator YAML would more naturally be considered `false`, not `true`. Secondly, an "opt-in" approach I think is the most prudent choice here, especially for highly-nested or related resources. Consider, for example, the case of API Gateway v2 where you have a top-level `Api` resource that has many related sub-resources under it, for example `Stage`, `Authorizer`, `Deployment`, etc. We don't (can't?) allow those sub-resources to be adopted because those sub-objects depend on the existence of a parent resource (an `Api`), and since our AdoptedResource struct only allows for singular primary key identifiers, adoption of dependent related objects isn't possible.

Reply:
I would believe (and that's based on naivety) that the majority of AWS resources are able to be adopted without any issue. Setting the default to `false` would require each service team to manually enable each of their resources - the majority of which would work without any generator modifications. For the sake of compatibility, enabling adoption for as many resources as possible would be preferable to customers hoping to adopt in the future.